websocket/platform/awscdk.w

```diff
@@ -0,0 +1,166 @@
+      if event.requestContext.routeKey == routeKey {
+        handler(event.requestContext.connectionId);
+      }
+
+      return {
+        statusCode: 200,
+        body: "ack"
+      };
+    })) as "on connect";
+
+    this.addRoute(onConnectFunction, {
+      routeKey: routeKey,
+    });
+  }
+  pub onDisconnect(handler: inflight(str): void): void {
+    let routeKey = "$disconnect";
+    let onDisconnectFunction = new cloud.Function(unsafeCast(inflight (event: awsapi.WebSocketAwsRequest): awsapi.WebSocketAwsResponse => {
+      if event.requestContext.routeKey == routeKey {
+        handler(event.requestContext.connectionId);
+      }
+
+      return {
+        statusCode: 200,
+        body: "ack"
+      };
+    })) as "on disconnect";
+
+    this.addRoute(onDisconnectFunction, {
+      routeKey: routeKey,
+    });
+  }
+  pub onMessage(handler: inflight(str, str): void): void {
+    let routeKey = "$default";
+    let onMessageFunction = new cloud.Function(unsafeCast(inflight (event: awsapi.WebSocketAwsRequest): awsapi.WebSocketAwsResponse => {
+      if event.requestContext.routeKey == routeKey {
+        handler(event.requestContext.connectionId, event.body);
+      }
+
+      return {
+        statusCode: 200,
+        body: "ack"
+      };
+    })) as "on message";
+
+    this.addRoute(onMessageFunction, {
+      routeKey: routeKey,
+    });
+  }
+
+  pub initialize() {}
+
+  pub addRoute(handler: cloud.Function, props: commons.RouteOptions): void {
+    if let lambda = aws.Function.from(handler) {
+      let functionArn = lambda.functionArn;
+      
+      this.role.addToPolicy(
+        new awscdk.aws_iam.PolicyStatement(
+          effect: awscdk.aws_iam.Effect.ALLOW,
+          resources: [functionArn],
+          actions: ["lambda:InvokeFunction"]
+        )
+      );
+
+      let integrationUri = awscdk.Stack.of(this).formatArn({
+        service: "apigateway",
+        account: "lambda",
+        resource: "path/2015-03-31/functions",
+        resourceName: "{functionArn}/invocations",
+      });
+      
+      let integration = new awscdk.aws_apigatewayv2.CfnIntegration(
+        apiId: this.api.attrApiId,
+        integrationType: "AWS_PROXY",
+        integrationUri: integrationUri,
+        credentialsArn: this.role.roleArn,
+      ) as "{props.routeKey}Integration";
+
+      let route = new awscdk.aws_apigatewayv2.CfnRoute(
+        apiId: this.api.attrApiId,
+        routeKey: props.routeKey,
+        authorizationType: "NONE",
+        target: "integrations/{integration.ref}",
+      ) as "{props.routeKey}Route";
+
+      this.deployment.addDependency(route);
+    }
+  }
+
+  pub inflight url(): str {
+    return this.invokeUrl;    
+  }
+
+  extern "../inflight/websocket.aws.mts" static inflight _postToConnection(endpointUrl: str, connectionId: str, message: str): void;
+  pub inflight sendMessage(connectionId: str, message: str) {
+    WebSocket_awscdk._postToConnection(this.callbackUrl, connectionId, message);
+  }
```

Comment:
@MarkMcCulloh this makes me wonder if perhaps we can change `extern` to be something like this:

```js
  pub inflight sendMessage(connectionId: str, message: str) {
    extern "../inflight/websocket.aws.mts" _postToConnection(endpointUrl: str, connectionId: str, message: str): void;
    
    _postToConnection(this.callbackUrl, connectionId, message);
  }
```